student/tests/views/test_registration_views.py

```diff
@@ -85,3 +88,45 @@ def test_registration_wizard_form_new_user(batch, existing_user):
     })
 
     assert User.objects.all().count() == 2
+
+def test_payment_preview_get_render_method_discount_logic():
+    course = Course.objects.create(name=settings.CODING_BASICS)
+    batch = Batch.objects.create(
+        course=course,
+        start_date=datetime.date.today(),
+        end_date=datetime.date.today() + datetime.timedelta(days=1),
+        capacity=1,
+        sections=1,
+    )
+    section = Section.objects.create(
+        batch=batch,
+        number=1,
+        capacity=1
+    )
+    coupon_effect = CouponEffect.objects.create(
+        couponable_type='Course',
+        couponable_id=1,
+        discount_type='dollars',
+        discount_amount=10
+    )
+    coupon = Coupon.objects.create(
+        start_date=make_aware(datetime.datetime.now()),
+    )
+    coupon.effects.set([coupon_effect])
+    registration = Registration.objects.create(
+        course=course,
+        batch=batch,
+        first_name='FirstName',
+        last_name='LastName',
+        email='user@email.com',
+        country_of_residence='SG',
+        referral_channel='word_of_mouth',
+        referral_code=coupon.code
+    )
+
+    registration = Registration.objects.get(pk=1)
```

Comment:
pls put this above with the other 'arrange' statements! separate the action into it's own visual block.